code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm

```diff
@@ -188,7 +190,9 @@
 
 /datum/action/xeno_action/stealth/proc/damage_taken(mob/living/carbon/xenomorph/X, damage_taken)
 	SIGNAL_HANDLER
-	if(damage_taken > 15)
+	if(!stealth)
+		return
```

Comment:
Why add this check

Reply:
Also i dont see this signal eing unregistered anywhere

Reply:
Oh, yeah. It's supposed to be in the big remove_action unregister block, but whoever first added it forgot that and I didn't notice. Check's to prevent proc call overhead because this's getting called whenever you get hit regardless of stealth activation.

Reply:
What? Why would it silently fail during stealth? This makes it so if you aren't in stealth, cancel_stealth() isn't called even if you pass the damage breakpoint.

Reply:
> 
> 
> What? Why would it silently fail during stealth? This makes it so if you aren't in stealth, cancel_stealth() isn't called even if you pass the damage breakpoint.

its still getting called, youre just not letting it do anything, which means is still doing proc overhead